sycl/test-e2e/Matrix/XMX8/joint_matrix_optional_kernel_feature.cpp

```diff
@@ -0,0 +1,127 @@
+// REQUIRES: matrix
+// REQUIRES: matrix-xmx8
```

Comment:
With settings like that, this test might be skipped for both DG2 and PVC. I think we need 2 cpp files and one hpp file, similar to other tests. With one cpp file to require 'matrix' to run on PVC and another cpp file to require 'matrix-xmx8' to run on DG2 and hpp file to avoid duplication.

Also could you simplify the test below as much as possible, since the point is to only check exception, right? So we don't care about matrix_compare, matrix_multiply_ref. We don't need matrix D, etc....

Reply:
Thanks, didn't know that